File: src/main/java/com/agriculture/modules/camera/entity/Camera.java
package com.agriculture.modules.camera.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import lombok.Getter;
import lombok.Setter;

/**
 * <p>
 * 摄像头表
 * </p>
 *
 * @author agriculture-team
 * @since 2026-06-09
 */
@Getter
@Setter
@TableName("camera")
public class Camera implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * 摄像头UUID
     */
    @TableId(value = "id", type = IdType.AUTO)
    private String id;

    /**
     * 摄像头名称
     */
    @TableField("name")
    private String name;

    /**
     * RTSP流地址
     */
    @TableField("rtsp_url")
    private String rtspUrl;

    /**
     * RTSP子码流地址
     */
    @TableField("rtsp_url_sub")
    private String rtspUrlSub;

    /**
     * 经度
     */
    @TableField("location_x")
    private BigDecimal locationX;

    /**
     * 纬度
     */
    @TableField("location_y")
    private BigDecimal locationY;

    /**
     * 朝向角度
     */
    @TableField("direction")
    private BigDecimal direction;

    /**
     * 抓拍分辨率
     */
    @TableField("capture_resolution")
    private String captureResolution;

    /**
     * 抓拍JPEG质量(1-100)
     */
    @TableField("capture_quality")
    private Integer captureQuality;

    /**
     * 断流重连间隔(秒)
     */
    @TableField("reconnect_interval")
    private Integer reconnectInterval;

    /**
     * 状态: ONLINE/OFFLINE/FAULT
     */
    @TableField("status")
    private String status;

    /**
     * 最后抓拍时间
     */
    @TableField("last_frame_at")
    private LocalDateTime lastFrameAt;

    /**
     * 最近一次在线时间
     */
    @TableField("last_online_at")
    private LocalDateTime lastOnlineAt;

    /**
     * 创建时间
     */
    @TableField("created_at")
    private LocalDateTime createdAt;

    /**
     * 更新时间
     */
    @TableField("updated_at")
    private LocalDateTime updatedAt;

    /**
     * 逻辑删除
     */
    @TableField("deleted")
    private Byte deleted;

    /**
     * 覆盖网格ID列表（非数据库字段）
     */
    @TableField(exist = false)
    private List<String> coverageGrids;
}

File: frontend/src/api/camera.ts
/**
 * 摄像头模块 API
 * 对接后端 /api/camera 接口
 */
import { request } from './request'
import type { PageResult } from './request'

const BASE = '/api/camera'

/** 摄像头实体 */
export interface CameraVO {
  id: string
  name: string
  rtspUrl: string
  rtspUrlSub: string | null
  locationX: number
  locationY: number
  direction: number
  captureResolution: string
  captureQuality: number
  reconnectInterval: number
  status: 'ONLINE' | 'OFFLINE' | 'FAULT'
  lastFrameAt: string | null
  lastOnlineAt: string | null
  companyId: string
  createdAt: string
  updatedAt: string
}

/** 获取摄像头列表（分页） */
export async function fetchCameras(params: {
  status?: string
  keyword?: string
  page?: number
  size?: number
} = {}): Promise<PageResult<CameraVO>> {
  const query = new URLSearchParams()
  if (params.status) query.set('status', params.status)
  if (params.keyword) query.set('keyword', params.keyword)
  if (params.page) query.set('page', String(params.page))
  if (params.size) query.set('size', String(params.size))
  return request<PageResult<CameraVO>>(`${BASE}/list?${query.toString()}`)
}

/** 更新摄像头 */
export async function updateCamera(id: string, data: {
  name?: string
  rtspUrl?: string
  rtspUrlSub?: string
  locationX?: number
  locationY?: number
  direction?: number
  coverageGrids?: string[]
  captureResolution?: string
  captureQuality?: number
  reconnectInterval?: number
}): Promise<void> {
  return request<void>(`${BASE}/${id}`, {
    method: 'PUT',
    body: JSON.stringify(data),
  })
}

/** 删除摄像头 */
export async function deleteCamera(id: string): Promise<void> {
  return request<void>(`${BASE}/${id}`, { method: 'DELETE' })
}

/** 创建摄像头 */
export async function createCamera(data: {
  name: string
  rtspUrl: string
  rtspUrlSub?: string
  locationX?: number
  locationY?: number
  direction?: number
  coverageGrids?: string[]
  captureResolution?: string
  captureQuality?: number
  reconnectInterval?: number
}): Promise<{ id: string }> {
  return request<{ id: string }>(BASE, {
    method: 'POST',
    body: JSON.stringify(data),
  })
}
